src/runtimes/node/bundlers/esbuild/bundler.ts

```diff
@@ -147,6 +155,31 @@ let require=___nfyCreateRequire(import.meta.url);
       srcFile,
       outputExtension,
     })
+
+    for (const [filename] of Object.entries(metafile.outputs)) {
+      const content = await readFile(filename, { encoding: 'utf-8' })
+
+      const updated = content.replace(
+        `
+var __glob = (map) => (path) => {
```

Comment:
If there is a symbol called `__glob` defined in user space (whether by a function directly or by a dependency), esbuild will rename this symbol to one that is available, following the format `__glob2`, `__glob3`, etc. If that happens, we won't match and will fail to patch this function.

We could potentially solve this by passing a regular expression to `content.replace`, probably at the cost of a more expensive matching.

This feels like an edge case though, so I understand if you don't want to address it.

Reply:
you could use a variable name that is so unlikely to be used including emojis. `__glob💩-thingy`

Reply:
Could you share some code where this happens? I've added a test on this in [2dd12bb](https://github.com/netlify/zip-it-and-ship-it/pull/1532/commits/2dd12bb17043b3521a81123c7c5b3c4f0aa6b5b9), and it doesn't break. Taking a look at the output, it looks like it renames the user-defined `__glob`, not its own one:

<img width="952" alt="Screenshot 2023-08-24 at 13 02 36" src="https://github.com/netlify/zip-it-and-ship-it/assets/14912729/484f66c8-25fa-4f34-b6c7-7e664bfa4425">

Reply:
Ah, I was mistaken. I was convinced that esbuild would keep the name of the user symbol and rename the shim, but it's the other way around. Ignore me!